src/zenohRpcClient.cpp

```diff
@@ -126,90 +125,92 @@ std::future<UPayload> ZenohRpcClient::invokeMethod(const UUri &uri,
+    z_bytes_map_insert_by_alias(&map, z_bytes_new("header"), headerBytes);
+    z_bytes_map_insert_by_alias(&map, z_bytes_new("payload"), payloadBytes);
 
-   *channel = zc_reply_fifo_new(16);
+    z_owned_reply_channel_t *channel = new z_owned_reply_channel_t;
+    *channel = zc_reply_fifo_new(16);
 
     z_get_options_t opts = z_get_options_default();
-
     opts.timeout_ms = requestTimeoutMs_;
-
-    opts.value.payload = (z_bytes_t){.len =  message.size(), .start = (uint8_t *)message.data()};
-
-    auto uuidStr = UuidSerializer::serializeToString(attributes.id());
+    opts.attachment = z_bytes_map_as_attachment(&map);
 
     if (0 != z_get(z_loan(session_), z_keyexpr(std::to_string(uriHash).c_str()), "", z_move(channel->send), &opts)) {
         spdlog::error("z_get failure");
+        z_drop(&map);
         return std::move(future);
-    }  
-    
+    }
+
     future = threadPool_->submit(handleReply, channel);
 
-    if (false == future.valid()) {
+    if (!future.valid()) {
         spdlog::error("failed to invoke method");
     }
-   
+
+    z_drop(&map);
     return future; 
 }
 
 UPayload ZenohRpcClient::handleReply(z_owned_reply_channel_t *channel) {
-
     z_owned_reply_t reply = z_reply_null();
-    UPayload response (nullptr, 0, UPayloadType::VALUE);
-        
-    for (z_call(channel->recv, &reply); z_check(reply); z_call(channel->recv, &reply)) {
+    UPayload response(nullptr, 0, UPayloadType::VALUE);
 
+    for (z_call(channel->recv, &reply); z_check(reply); z_call(channel->recv, &reply)) {
         if (z_reply_is_ok(&reply)) {
             z_sample_t sample = z_reply_ok(&reply);
-            z_owned_str_t keystr = z_keyexpr_to_string(sample.keyexpr);
 
-            z_drop(z_move(keystr));
+            // Attachment handling and TLV extraction
+            std::optional<std::unordered_map<Tag, TLV>> allTlv;
+            if (z_check(sample.attachment)) {
+                z_bytes_t index = z_attachment_get(sample.attachment, z_bytes_new("header"));
+                if (z_check(index)) {
+                    spdlog::info("Attachment: value = '%.*s'", (int)index.len, index.start);
+                    allTlv = MessageParser::instance().getAllTlv(reinterpret_cast<const uint8_t*>(index.start), index.len);
+                }
+            } else {
+                allTlv = MessageParser::instance().getAllTlv(reinterpret_cast<const uint8_t*>(sample.payload.start), sample.payload.len);
+            }
 
-            auto tlvVector = MessageParser::getAllTlv(sample.payload.start, sample.payload.len);
-     
-            if (false == tlvVector.has_value()) {
+            if (!allTlv.has_value()) {
+                spdlog::error("MessageParser::getAllTlv failure");
+                continue;
+            }
 
-                spdlog::error("getAllTlv failure");
-                return response;
+            auto header = MessageParser::instance().getAttributes(allTlv.value());
```

Comment:
the payload is not part of the TLV anymore , but received separately same as in the pub-sub flow